src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -2635,9 +2638,9 @@ private Task SNIWritePacket(SNIHandle handle, SNIPacket packet, out uint sniErro
                 }
                 else
                 {
-                    SqlClientEventSource.Log.TryTraceEvent("<sc.TdsParser.WritePacket|Info> write async returned error code {0}", (int)sniError);
+                    SqlClientEventSource.Log.TryTraceEvent("TdsParserStateObject.SNIWritePacket | Info | State Object Id {0}, Write async returned error code {1}", _objectID, (int)sniError);
                     AddError(_parser.ProcessSNIError(this));
-                    ThrowExceptionAndWarning(callerHasConnectionLock, false);
+                    ThrowExceptionAndWarning(callerHasConnectionLock, asyncClose);
```

Comment:
I think this was a bug introduced in [1766](https://github.com/dotnet/SqlClient/pull/1766/files#diff-189db7b3c65094f064020580e11301eaefbaa3a5d6f5e6616b4c9dfd14e69117R3607).